src/dev/flang/ast/AbstractMatch.java

```diff
@@ -128,12 +128,7 @@ public void visitExpressions(ExpressionVisitor v)
    */
   private AbstractType typeFromCases()
   {
-    AbstractType result = Types.resolved.t_void;
-    for (var c: cases())
-      {
-        var t = c.code().typeForInferencing();
-        result = result == null || t == null ? null : result.union(t);
-      }
+    var result = Expr.union(cases().map2(x -> x.code()), false);
```

Comment:
Sorry, missed this in the original review: Why is the `ignoreUnkown` here set to `false`? For `if` this is `true` and I do not see the fundamental difference between `match` and `if`. 
If it could be `true` as well we could get rid of this argument altogether!